packages/back-end/src/controllers/features.ts

```diff
@@ -542,10 +635,28 @@ export async function postFeaturePublish(
   req.checkPermissions("manageFeatures", feature.project);
 
   const revision = await getRevision(org.id, feature.id, parseInt(version));
+  const reviewStatuses = [
+    "pending-review",
+    "changes-requested",
+    "draft",
+    "approved",
+  ];
+  if (adminOverride && org.settings?.requireReviews) {
+    req.checkPermissions("bypassApprovalChecks");
+  }
   if (!revision) {
     throw new Error("Could not find feature revision");
   }
-  if (revision.status !== "draft") {
+  if (
+    (!adminOverride && org.settings?.requireReviews) ||
+    revision.status !== "approved"
+  ) {
+    throw new Error("needs review before publishing");
+  }
+  if (
+    !org.settings?.requireReviews &&
+    !reviewStatuses.includes(revision.status)
```

Comment:
TS seemed to have a weird bug that was saying it was unintentional. because of checking approved above